### PR TITLE
V0.1.5

### PR DESCRIPTION
**Pull request by mtravers**
```
a6ff46e  add truncate-string, divide-with
c97199a  Add some walk abstractions
0d4d44d  -csv fns to parallel -tsv, add homedir expansion
4a03e26  add median
b75639f  add interacalate and random-elements
1fac8f7  list-dir-recursive ignores .dotfiles, remove in-place version of process-file-lines
8c1e786  map-keys* and dehumanize
8de0014  add ensure-file, following-elt, preceding-elt
161eeff  master add cljcore/invoke-private

```

### CODE REVIEW

File: docs/org.candelbio.multitool.clj-testing.html
Comment (kenrestivo) [inline docs/org.candelbio.multitool.clj-testing.html:1]: Would be great if GH allowed this to be excluded from git completely, or at least excluded from diffs/PRs.

File: src/cljc/org/candelbio/multitool/core.cljc [hunk 1]
```diff
@@ -447,17 +459,25 @@
   [seq]
   (concat seq (repeat nil)))
 
+;;; I thought split-with did this, but no. Surely this in core?
+(defn divide-with
+  [p coll]
+  (let [groups (group-by (comp true? p) coll)]
+    [(get groups true) (get groups false)]))
+
 (defn mapf
```
Comment (kenrestivo) [inline src/cljc/org/candelbio/multitool/core.cljc:468]: Like keep from core, but this uses nullish? where keep just checks for clojure falseyness

File: src/cljc/org/candelbio/multitool/core.cljc [hunk 2]
```diff
@@ -471,6 +491,18 @@
     (doall thing)
     thing))
 
+(defn following-elt
+  [elt seq]
+  (cond (empty? seq) nil
+        (= (first seq) elt) (second seq)
+        :else (following-elt elt (rest seq))))
```
Comment (kenrestivo) [inline src/cljc/org/candelbio/multitool/core.cljc:498]: No TCO in Clojure, might blow stack

File: src/cljc/org/candelbio/multitool/core.cljc [hunk 3]
```diff
@@ -711,6 +751,7 @@
   [f chunk-size l]
   (mapcat f (partition-all chunk-size l)))
 
+(macros/deftime
```
Comment (kenrestivo) [inline src/cljc/org/candelbio/multitool/core.cljc:754]: The indenting looks odd, might want to run indent-buffer in emacs on this file.

File: src/cljc/org/candelbio/multitool/core.cljc [hunk 4]
```diff
@@ -800,11 +842,25 @@
   [f hashmap]
   (reduce-kv (fn [acc k v] (assoc acc (f k) v)) {} hashmap))
 
+(defn map-keys-recursive
+  "Map f over the keys of hashmap, and any nested hashmaps"
+  [f hashmap]
+  (if (map? hashmap)
+    (reduce-kv (fn [acc k v] (assoc acc (f k) (map-keys-recursive f v))) {} hashmap)
```
Comment (kenrestivo) [inline src/cljc/org/candelbio/multitool/core.cljc:849]: TCO/stack on JVM/JS.

**Review by kenrestivo — APPROVED** [review]

File: src/cljc/org/candelbio/multitool/core.cljc (hunk 2 above)
Comment (mtravers) [reply src/cljc/org/candelbio/multitool/core.cljc:498]: I rewrote some of those to use loop, but in general its not that much of a problem if you are working with reasonable sized seqs, if they are really so large that they will blow out the stack you probably don't want to be using naive fns like this, stack-safe or not,

**Pull request merged**